classes/particles/fallback_particles.gd

```diff
@@ -0,0 +1,77 @@
+extends GPUParticles2D
+## Compatibility class that swaps GPU particles out for CPU particles on devices that don't support it.
+## Might be worth making an engine proposal for this kind of thing, since this is somewhat hacky.
+
+## The replacement CPUParticles2D node.
+var _replacement: CPUParticles2D = null
+
+## If true, _set() and _get() will not redirect to the replacement.
+var _dont_redirect: bool = false
+
+
+func _ready():
+	# Check if the device is missing a video adapter (GPU)
+	if OS.get_video_adapter_driver_info().is_empty():
+		# Temporarily disable redirection so we can work.
+		_dont_redirect = true
+		
+		# Create a CPU node and copy properties over.
+		_replacement = CPUParticles2D.new()
+		_copy_properties()
+		
+		# Replace this node.
+		replace_by.call_deferred(_replacement)
+		
+		# Rename the replacement to hide the change.
+		_replacement.name = name
+		
+		# Re-enable redirection.
+		_dont_redirect = false
+
+
+## Copy properties from the current node to the replacement.
+func _copy_properties() -> void:
+	_replacement.convert_from_particles(self)
+	
+	for property in get_property_list():
+		var prop_name = property.name
+		if property.usage != 6:
+			continue
```

Comment:
It's `PROPERTY_USAGE_STORAGE | PROPERTY_USAGE_EDITOR`. Flags that ensure that the property is editor-accessible and also stored in the scene file. If it's not both of those, then there's no point copying it over. I should probably use those instead of magic numbers, though.